vividus-plugin-web-app/src/test/java/org/vividus/proxy/steps/ProxyStepsTests.java

```diff
@@ -137,35 +144,96 @@ void checkHarEntryExistenceWithHttpMethodAndUrlPatternNoCalls() throws IOExcepti
             return map.size() == 2
-                    && map.containsKey(key1)
-                    && value1.equals(map.get(key1))
-                    && map.containsKey(key2)
-                    && value2.equals(map.get(key2));
+                    && map.containsKey(KEY1)
+                    && VALUE1.equals(map.get(KEY1))
+                    && map.containsKey(KEY2)
+                    && VALUE2.equals(map.get(KEY2));
+        }));
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void checkCaptureRequestDataFromHarEntry() throws IOException
+    {
+        ProxySteps spy = spy(proxySteps);
+        HarEntry harEntry = mock(HarEntry.class);
+        doReturn(List.of(harEntry)).when(spy).checkNumberOfRequests(HTTP_METHOD, URL_PATTERN,
+                ComparisonRule.EQUAL_TO, 1);
+        HarResponse harResponse = mock(HarResponse.class);
+        when(harEntry.getResponse()).thenReturn(harResponse);
+        HarRequest harRequest = mock(HarRequest.class);
+        when(harEntry.getRequest()).thenReturn(harRequest);
+        HarQueryParam firstPair = new HarQueryParam();
+        firstPair.setName(KEY1);
+        firstPair.setValue(VALUE1);
+        HarQueryParam secondPair = new HarQueryParam();
+        secondPair.setName(KEY2);
+        secondPair.setValue(VALUE2);
+        when(harRequest.getQueryString())
+                .thenReturn(List.of(firstPair, secondPair));
+        String mimeType = "mimeType";
+        String text = "text";
+        String comment = "comment";
+        HarPostData postData = new HarPostData();
+        postData.setMimeType(mimeType);
+        postData.setText(text);
+        postData.setComment(comment);
+        HarPostDataParam postDataParam1 = new HarPostDataParam();
+        HarPostDataParam postDataParam2 = new HarPostDataParam();
+        postData.setParams(List.of(postDataParam1, postDataParam2));
+        postDataParam1.setName(KEY1);
+        postDataParam1.setValue(VALUE1);
+        postDataParam2.setName(KEY2);
+        postDataParam2.setValue(VALUE2);
+        when(harRequest.getPostData()).thenReturn(postData);
+        when(harResponse.getStatus()).thenReturn(200);
+        Set<VariableScope> variableScopes = Set.of(VariableScope.SCENARIO);
+        spy.captureRequestAndSaveRequestData(HTTP_METHOD, URL_PATTERN, variableScopes, VARIABLE_NAME);
+        verify(bddVariableContext).putVariable(eq(variableScopes), eq(VARIABLE_NAME), argThat(value ->
+        {
+            Map<String, Map<String, String>> map = (Map<String, Map<String, String>>) value;
+            Map<String, String> queryMap = map.get("queryMap");
+            Map<String, String> requestBodyMap = map.get("requestBodyMap");
+            Map<String, String> requestBodyParamsMap = map.get("requestBodyParamsMap");
+            Map<String, String> responseStatusMap = map.get("responseStatusMap");
+            String status = "status";
+            return map.size() == 4
```

Comment:
use Assertions.assertAll(...)
In the current implementation, I have to debug and compare manually to understand what is not matching